diplom.py

```diff
@@ -40,44 +59,24 @@ def parse(line):
     multi_class = 'multinomial', max_iter = 100)
 print('LogisticRegression score: %f'
       % logistic.fit(train_X, train_y).score(test_X, test_y))
-Predicted_probability_distribution = logistic.predict_proba(test_X)
-#print logistic.classes_
+predicted_probability_distribution = logistic.predict_proba(test_X)
+n_of_classes = len(logistic.classes_)
+list_of_classes = logistic.classes_
+MAP = mean_average_precision (predicted_probability_distribution, 
+    n_of_classes, 
+    list_of_classes, 
+    test_y, 
+    dataset_size)
 
-AP = 0
-k = -1
-for row in Predicted_probability_distribution:
-    k += 1
-    r =[]
-    for i in xrange(0, len(logistic.classes_)):
-        r.append([row[i], logistic.classes_[i]])
-    r.sort(reverse=True)    
-    for j in range (0, 5):
-        if r[j][1] == test_y[k]:
-            AP += 1.0/(j+1)
-MAP = AP / TEST_DATASET_SIZE
-print (MAP)
+print MAP 
```

Comment:
Please output error on train and test datasets. This way you would be able to check basic expectation that your algorithm performs better on the data it already saw than on the new data. BTW, you can respond to my comments here.